File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/NotesModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DanpheEMR.ServerModel.ClinicalModels;


namespace DanpheEMR.ServerModel
{
    public class NotesModel
    {
        [Key]
        public int NotesId { get; set; }
        public int PatientVisitId { get; set; }
        public int PatientId { get; set; }
        public int ProviderId { get; set; }
        public int TemplateId { get; set; }
        public int? SecondaryDoctorId { get; set; }
        public int? NoteTypeId { get; set; }
        public string TemplateName { get; set; }
        public int? FollowUp { get; set; }
        public string FollowUpUnit { get; set; }
        public string Remarks { get; set; }
        public bool? IsPending { get; set; }

        public int? CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public string ICDSelected { get; set; }

        [NotMapped]
        public DateTime? ReceivedOn { get; set; }

        [NotMapped]
        public string VisitCode { get; set; }
        [NotMapped]
        public DateTime? VisitDate {get;set;}
        [NotMapped]
        public string ReferredBy { get; set; }
        [NotMapped]
        public virtual FreeTextNoteModel FreeTextNote { get; set; }
        
        [NotMapped]
        public virtual ProcedureNoteModel ProcedureNote { get; set; }
        [NotMapped]
        public virtual ProgressNoteModel ProgressNote { get; set; }
        [NotMapped]
        public virtual EmergencyNoteModel EmergencyNote { get; set; }
        //[NotMapped]
        //public virtual HistoryAndPhysicalNoteModel HistoryAndPhysicalNote { get; set; }        
        [NotMapped]
        public virtual DischargeSummaryModel DischargeSummaryNote { get; set; }

        [NotMapped]
        public virtual SubjectiveNoteModel SubjectiveNote { get; set; }
        [NotMapped]
        public virtual ObjectiveNoteModel ObjectiveNote { get; set; }
        [NotMapped]
        public virtual PrescriptionNotesModel ClinicalPrescriptionNote { get; set; }
        [NotMapped]
        public virtual List<ClinicalDiagnosisModel> AllIcdAndOrders { get; set; }

        public virtual List<ClinicalDiagnosisModel> RemovedIcdAndOrders { get; set; }
    }

}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical-notes/shared/notes.model.ts
import { EmergencyNotesModel } from "./emergency-note.model";
import { ProcedureNotesModel } from "./procedure-note.model";
import { SubjectiveNotesModel } from "./subjective-note.model";
import { ObjectiveNotesModel } from "./objective-notes.model";
import { FreeTextNotesModel } from "./freetext.model";
import { AssessmentAndPlanModel } from "./assessment-and-plan.model";
import { IcdWithOrdersViewModel } from "../../clinical/shared/all-icd-with-orders.viewmodel";

import * as moment from "moment/moment"
import { DischargeSummary } from "../../adt/shared/discharge-summary.model";
import { ProgressNotesModel } from "./progress-note.model";
import { ClinicalPrescriptionNotesModel } from "./clinical-prescription-note.model";
export class NotesModel {
  public NotesId: number = 0;
  public PatientVisitId: number = null;
  public SecondaryDoctorId: number = null;
  public PatientId: number = null;
  public ProviderId: number = null;
  public TemplateId: number = null;
  public TemplateName: string = null;
  public NoteTypeId: number = null;
  public Remarks: string = null;
  public IsPending: boolean = null;
  public Date: string = moment().format("YYYY-MM-DD");
  public CreatedBy: number = null;
  public ModifiedBy: number = null;
  public CreatedOn: string = null;
  public ModifiedOn: string = null;
  public FollowUp: number = null;
  public FollowUpUnit: string = "day";
  public FreeTextNote: FreeTextNotesModel = new FreeTextNotesModel();
  public EmergencyNote: EmergencyNotesModel = new EmergencyNotesModel();
  public ProcedureNote: ProcedureNotesModel = new ProcedureNotesModel();
  public ProgressNote: ProgressNotesModel = new ProgressNotesModel();
  public DischargeSummaryNote: DischargeSummary = new DischargeSummary();

  //models for history and physical note
  public SubjectiveNote: SubjectiveNotesModel = new SubjectiveNotesModel();
  public ObjectiveNote: ObjectiveNotesModel = new ObjectiveNotesModel();
  public ClinicalDiagnosis: AssessmentAndPlanModel = new AssessmentAndPlanModel();
  public ClinicalPrescriptionNote: ClinicalPrescriptionNotesModel = new ClinicalPrescriptionNotesModel();

  //for assessment and plan - orders preparation
  public AllIcdAndOrders: Array<IcdWithOrdersViewModel> = [];

  public RemovedIcdAndOrders: Array<IcdWithOrdersViewModel> = [];


  //used in receive note
  public ReceivedOn: string = null;

}
